src/main/java/org/folio/edge/controller/exception/ExceptionHandlerController.java

```diff
@@ -77,4 +83,17 @@ public Error handleException(Exception e) {
     return new ResponseEntity<>(new Error().error(INVALID_REQUEST_ERROR_TYPE), HttpStatus.valueOf(e.getStatus()));
   }
 
+  @ExceptionHandler(FeignException.class)
+  public ResponseEntity<InnReachResponseDTO> handleFeignStatusException(FeignException feignException) {
+    var status = HttpStatus.resolve(feignException.status());
+    String body = feignException.contentUTF8();
+    InnReachResponseDTO innReachResponseDTO = null;
+    try {
+      innReachResponseDTO = objectMapper.readValue(body, InnReachResponseDTO.class);
+    } catch (JsonProcessingException e) {
+      e.printStackTrace();
+    }
```

Comment:
The `innReachResponseDTO` in this case is not initialized, which will lead to null response. Prepare an error manually as a fall-back in catch block